### ISSUE TITLE
[BUG] Update from 1.4.2 to 1.5.3 illegal offset type in GenericMetadata.php

### ISSUE THREAD

**Issue by entering** [FriendsOfSymfony/FOSRestBundle#973]
```
Warning: Illegal offset type in isset or empty in /vagrant/vendor/symfony/symfony/src/Symfony/Component/Validator/Mapping/GenericMetadata.php line 216
500 Internal Server Error - ContextErrorException
```

symfony version: v2.5.3

The code:

```php
    /**
     * @QueryParam(
     *      name="letter",
     *      nullable=false,
     *      requirements="[A-Z\#]",
     *      default="A",
     *      strict=true,
     *      description="First letter of the Accommodation Provider's name to filter by"
     * )
     *
     * @param Request $request
     * @param string $letter
     *
     * @return Response
     */
    public function listAction(Request $request, $letter)
{
...
}
```

And the routing:
```yml
accommodation_provider_list:
    pattern:  /accommodation-providers
    defaults: { _controller: accommodation_bundle.accommodation_provider.controller:listAction }
    methods: [POST, GET]
```

**Comment by lsmith77** [comment]
can you investigate this a but or submit a test a failing test case? there has been some refactoring in this area

**Comment by entering** [comment]
@lsmith77 yes, I just opened the issue to avoid getting "lost", I will investigate and come at least with a failing test case

**Comment by lsmith77** [comment]
ping

**Comment by lsmith77** [comment]
did you also open a ticket on Symfony core?

/cc @webmozart

**Comment by mlimaloureiro** [comment]
hello, i was investigating a little bit this issue, and the problem is with symfony LegacyValidator. Somehow, it is expecting an object instead of a value. Anyway this is fixed in 2.5.10, didn't investigate any further since updating symfony fixed the problem

**Comment by lsmith77** [comment]
maybe it was fixed by https://github.com/symfony/Validator/commit/b90adb60265bec674a0a5ac12bff95d93f7d78f1

I guess then we could in theory add a conflict to lower versions. reading the commit info it would then be an issue with 2.5 versions up to exclusing 2.5.5. it never affected versions below 2.5 of course and also didn't affect 2.6 versions.
